lib/stat_tracker.rb

```diff
@@ -8,22 +8,28 @@ class StatTracker
   attr_reader :games, :teams, :game_teams
 
   def self.from_csv(locations)
-    games = []
-    teams = []
-    game_teams = []
-    
-    CSV.foreach(locations[:games], headers: true, header_converters: :symbol) do |row|
-      games << Game.new(row.to_h)
-    end
-    CSV.foreach(locations[:teams], headers: true, header_converters: :symbol) do |row|
-      teams << Team.new(row.to_h)
-    end
-    CSV.foreach(locations[:game_teams], headers: true, header_converters: :symbol) do |row|
-      game_teams << GameTeam.new(row.to_h)
-    end
+    # The from_csv method initializes StatTracker object
+    # with the arrays of Game, Team, and GameTeam objects created from the CSV files. 
+    # The read_and_process_csv method abstracts the common logic of reading and creating objects from a CSV file.
+    # Calls read_and_process_csv for each type of CSV file 
+    # (games, teams, game_teams), passing the appropriate class (Game, Team, GameTeam) to construct the objects.
+    games = read_and_process_csv(locations[:games], Game)
```

Comment:
The read_and_process_csv is a creative way to approach this (reading and shoveling the data). It's doing a lot!